homeassistant/components/coned/sensor.py

```diff
@@ -0,0 +1,92 @@
+"""Platform for sensor integration."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from coned import Meter
+from coned import MeterError
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import ENERGY_KILO_WATT_HOUR
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_METER_NUMBER = "meter_number"
+
+SCAN_INTERVAL = timedelta(minutes=15)
+
+SENSOR_NAME = "ConEdison Current Energy Usage"
+SENSOR_ICON = "mdi:counter"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Required(CONF_METER_NUMBER): cv.string})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the sensor platform."""
+
+    meter_number = config[CONF_METER_NUMBER]
+
+    try:
+        meter = Meter(meter_number)
+
+    except MeterError:
+        _LOGGER.error("Unable to create ConEd meter")
+        return
+
+    add_entities([CurrentEnergyUsageSensor(meter)], True)
+
+    _LOGGER.debug("ConEd meter_number = %s", meter_number)
+
+
+class CurrentEnergyUsageSensor(Entity):
+    """Representation of the sensor."""
+
+    def __init__(self, meter):
+        """Initialize the sensor."""
+        self._state = None
+        self._available = None
+        self.meter = meter
+
+    @property
+    def unique_id(self):
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return self.meter.meter_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return SENSOR_NAME
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return SENSOR_ICON
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
```

Comment:
We should add `def device_state_attributes()` with an attribution to the source of the data.

Reply:
Added